### PR TITLE
doc: release-notes: minor fixup to the CAN release notes for v3.2.0

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
Add a couple of minor fixups to the CAN related release notes for Zephyr v3.2.0 (fixed typo and removed duplicate entry).

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

### CODE REVIEW

**Review by fabiobaltieri — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

File: doc/releases/release-notes-3.2.rst [hunk 1]
```diff
@@ -293,7 +293,6 @@ Drivers and Sensors
   * The Zephyr network CAN bus driver, which provides raw L2 access to the CAN bus via a CAN
     controller driver, has been moved to :zephyr_file:`drivers/net/canbus.c` and can now be enabled
     using :kconfig:option:`CONFIG_NET_CANBUS`.
-  * STM32: Now supports dual CAN instances.
```
Comment (mmahadevan108) [inline doc/releases/release-notes-3.2.rst:296]: Why is this deleted

Comment (henrikbrixandersen) [reply doc/releases/release-notes-3.2.rst:296]: It's in the commit log and the description: "removed duplicate entry".

**Review by mmahadevan108 — APPROVED** [review]
